kytos/core/auth.py

```diff
@@ -44,6 +50,83 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+class UserController:
+    """UserController"""
+
+    # pylint: disable=unnecessary-lambda
+    def __init__(self, get_mongo=lambda: Mongo()):
+        self.mongo = get_mongo()
+        self.db_client = self.mongo.client
+        self.db = self.db_client[self.mongo.db_name]
+
+    def bootstrap_indexes(self):
+        """Bootstrap all topology related indexes."""
+        index_tuples = [
+            ("users", [("username", pymongo.ASCENDING)]),
+        ]
+        for collection, keys in index_tuples:
+            if self.mongo.bootstrap_index(collection, keys):
+                LOG.info(
+                    f"Created DB index {keys}, collection: {collection}"
+                )
+
+    def create_user(self, user_data):
+        """Create user to database"""
+        utc_now = datetime.datetime.utcnow()
+        try:
+            data = {
+                "username": user_data["username"],
+                "email": user_data["email"],
+                "password": user_data["password"],
+                "inserted_at": utc_now,
+                "updated_at": utc_now,
+            }
+            user = UserDoc(**data)
```

Comment:
Instead of trying to construct another dict, it's easier to maintain if you pass it directly to `UserDoc` and let pydantic do its validation, that way we don't need to deal with `KeyError`. For `inserted_at` and `updated_at` you can leverage MongoDB [`$setOnInsert` operator](https://github.com/kytos-ng/flow_manager/blob/master/controllers/__init__.py#L90)

Reply:
Could this work too? `user_data` is from the argument.
```
            utc_now = datetime.datetime.utcnow()
            user_data["inserted_at"] = utc_now
            user_data["updated_at"] = utc_now
            result = self.db.users.insert_one(UserDoc(**user_data).dict())
```

Reply:
Yes, @Alopalao, since here you have create and update on different methods that's ok too. On other NApps where we have upsert, then `$setOnInsert` helps in that case too. 

On this code here, though you're mutating a reference, this can lead to surprises for callers, although as we have here it's only being used in a payload for the request and that isn't re-used after calling this method so we won't have issues in the near team, what I'll recommend is to build build a new dict `{**user_data, **{"inserted_at": utc_now, {"updated_at": utc_now}` to make the intention clear that we're just taking a dict and building it to insert an user.

Reply:
Yeah, it is better.